### PR TITLE
Redact credentials from iOS TEE bridge logs

### PR DESCRIPTION
**Pull request by tomas-martins-crossmint**
This strips `apiKey`, `jwt`, and `encryptedOtp` out of log messages before any provider sees them. The TEE bridge logs were the leak that prompted this, but the redaction sits in `Logger` so every module gets it.

No change to the actual bridge behavior. The message sent to and from the webview is byte-for-byte the same; only what gets logged changed. The two bridge files are untouched.

OSLog privacy specifiers can't solve this. `%{private}@` and the `privacy:` interpolation parameter redact only inside the system log store, and never reach a provider that holds the message as a plain `String` and ships it over the network.

## Changes

- Add `CredentialScrubber`, matched on four patterns: JWT shape (`eyJ` plus two base64url segments), API key shape (`ck`/`sk` across all three environments), the key names `jwt`/`apiKey`/`encryptedOtp`, and the whole `authData`/`onboardingAuthentication` containers so a flat field added to either is covered without editing the file.
- Scrub in `Logger`, where all four levels funnel through one private method so a level added later can't skip it. String attribute values are scrubbed too; other `Encodable` values keep their type unless scrubbing changed something.
- Delete `SensitiveMessageRedactor` from the `Web` module, along with the redaction calls at the four bridge log sites.
- Add `MockLoggerProvider` under `Tests/LoggerTests/Mocks/` and drop the duplicate spy it replaces.

The API key prefixes are duplicated from `ApiKey.swift` rather than shared. `CrossmintService` depends on `Logger`, so importing it would create a cycle.

### CODE REVIEW

File: Sources/Web/WebViewCommunicationProxy.swift [hunk 1]
```diff
@@ -168,7 +168,7 @@ public class DefaultWebViewCommunicationProxy: NSObject, ObservableObject, WKScr
         }
 
         let script = "window.onMessageFromNative(\(jsonString));"
-        Logger.web.debug("Native >> Web: \(jsonString)")
+        Logger.web.debug("Native >> Web: \(SensitiveMessageRedactor.redactedLoggableString(from: messageData))")
```
Comment (afeight) [inline Sources/Web/WebViewCommunicationProxy.swift]: can we build this into the logger instead and get this for free in other places?

**Review by afeight — APPROVED** [review]
works but is more a band-aid than long-term
